### PR TITLE
En_Karebaba

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

**Review by AngheloAlf — APPROVED** [review]

File: src/overlays/actors/ovl_En_Karebaba/z_en_karebaba.c [hunk 1]
```diff
@@ -93,79 +128,619 @@ static DamageTable D_808F2E00 = {
...
+
+    if (this->unk_1EE != 0) {
+        this->unk_1EE--;
+    }
+
+    if (Animation_OnFrame(&this->skelAnime, 0.0f) || Animation_OnFrame(&this->skelAnime, 12.0f)) {
+        Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_MIZUBABA1_MOUTH);
+    }
+
+    if (this->collider2.base.acFlags & 2) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Karebaba/z_en_karebaba.c:311]:
```suggestion
    if (this->collider2.base.acFlags & AC_HIT) {
```

File: src/overlays/actors/ovl_En_Karebaba/z_en_karebaba.c [hunk 2]
```diff
@@ -93,79 +128,619 @@ static DamageTable D_808F2E00 = {
...
+    if (Animation_OnFrame(&this->skelAnime, 0.0f) || Animation_OnFrame(&this->skelAnime, 12.0f)) {
+        Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_MIZUBABA1_MOUTH);
+    }
+
+    phi_v0 = 20 - this->unk_1EE;
+    if (phi_v0 < 0) {
+        abs = -phi_v0;
+    } else {
+        abs = phi_v0;
+    }
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Karebaba/z_en_karebaba.c:351]:
```suggestion
   abs = ABS_ALT(phi_v0);
```

File: src/overlays/actors/ovl_En_Karebaba/z_en_karebaba.c [hunk 3]
```diff
@@ -93,79 +128,619 @@ static DamageTable D_808F2E00 = {
...
+    this->collider1.dim.radius = sCylinderInit2.dim.radius + (phi_v0 * 2);
+    this->actor.shape.rot.x = 0xC000 - (phi_v0 << 8);
+    this->actor.shape.rot.y += phi_v0 * 0x2C0;
+    this->actor.world.pos.y = (Math_SinS(this->actor.shape.rot.x) * -60.0f) + this->actor.home.pos.y;
+
+    sp28 = Math_CosS(this->actor.shape.rot.x) * 60.0f;
+    this->actor.world.pos.x = (Math_SinS(this->actor.shape.rot.y) * sp28) + this->actor.home.pos.x;
+    this->actor.world.pos.z = (Math_CosS(this->actor.shape.rot.y) * sp28) + this->actor.home.pos.z;
+
+    if (this->collider2.base.acFlags & 2) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Karebaba/z_en_karebaba.c:367]:
```suggestion
    if (this->collider2.base.acFlags & AC_HIT) {
```

File: src/overlays/actors/ovl_En_Karebaba/z_en_karebaba.c [hunk 4]
```diff
@@ -93,79 +128,619 @@ static DamageTable D_808F2E00 = {
...
+
+void func_808F1C84(EnKarebaba* this, GlobalContext* globalCtx) {
+    s32 i;
+    Vec3f sp78;
+    f32 temp_f20;
+    f32 temp_f22;
+    f32 temp_f24;
+
+    if (this->unk_1EC == 10) {
+        this->unk_1EE += -1;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Karebaba/z_en_karebaba.c:400]:
```suggestion
        this->unk_1EE--;
```

File: src/overlays/actors/ovl_En_Karebaba/z_en_karebaba.c [hunk 5]
```diff
@@ -93,79 +128,619 @@ static DamageTable D_808F2E00 = {
...
+
+void func_808F200C(EnKarebaba* this, GlobalContext* globalCtx) {
+    if (this->unk_1EC == 10) {
+        this->unk_1EE--;
+        if (this->unk_1EE == 0) {
+            func_808F1374(this, globalCtx);
+        }
+    } else {
+        if (Math_StepToF(&this->actor.scale.x, 0.0f, 0.0005f)) {
+            Item_DropCollectible(globalCtx, &this->actor.world.pos, 12);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Karebaba/z_en_karebaba.c:466]:
```suggestion
            Item_DropCollectible(globalCtx, &this->actor.world.pos, ITEM00_NUTS_1);
```

File: src/overlays/actors/ovl_En_Karebaba/z_en_karebaba.c [hunk 6]
```diff
@@ -93,79 +128,619 @@ static DamageTable D_808F2E00 = {
...
+            Math_StepToF(&this->actor.world.pos.y, this->actor.home.pos.y, 1.0f);
+        }
+        this->actor.scale.y = this->actor.scale.x;
+        this->actor.scale.z = this->actor.scale.x;
+    }
+}
+
+void func_808F20FC(EnKarebaba* this, GlobalContext* globalCtx) {
+    Actor_SetScale(&this->actor, 0.03f);
+    this->actor.shape.rot.x += -0x4000;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Karebaba/z_en_karebaba.c:479]:
```suggestion
    this->actor.shape.rot.x -= 0x4000;
```

File: src/overlays/actors/ovl_En_Karebaba/z_en_karebaba.c [hunk 7]
```diff
@@ -93,79 +128,619 @@ static DamageTable D_808F2E00 = {
...
+
+void func_808F21A4(EnKarebaba* this, GlobalContext* globalCtx) {
+    if (this->unk_1EE != 0) {
+        this->unk_1EE--;
+    }
+
+    if (Actor_HasParent(&this->actor, globalCtx) || (this->unk_1EE == 0)) {
+        func_808F238C(this);
+    } else {
+        Actor_PickUpNearby(&this->actor, globalCtx, 25);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Karebaba/z_en_karebaba.c:499]:
```suggestion
        Actor_PickUpNearby(&this->actor, globalCtx, GI_STICKS_1);
```

File: src/overlays/actors/ovl_En_Karebaba/z_en_karebaba.h [hunk 8]
```diff
@@ -7,11 +7,29 @@ struct EnKarebaba;
...
+    /* 0x018C */ Vec3s jointTable[8];
+    /* 0x01BC */ Vec3s morphTable[8];
+    /* 0x01EC */ u8 unk_1EC;
+    /* 0x01EE */ s16 unk_1EE;
+    /* 0x01F0 */ f32 unk_1F0;
+    /* 0x01F4 */ f32 unk_1F4;
+    /* 0x01F8 */ f32 unk_1F8;
+    /* 0x01FC */ Vec3f unk_1FC;
+    /* 0x0208 */ Vec3f unk_208[2];
+    /* 0x0220 */ Vec3f unk_220;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Karebaba/z_en_karebaba.h:29]:
```suggestion
    /* 0x01FC */ Vec3f unk_1FC[4];
```
From `func_800BE680`, this is suppose to be `limbPos[4]`

**Review by hensldm — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**